instance/tests/test_ansible.py

```diff
@@ -96,41 +98,105 @@ class AnsibleTestCase(TestCase):
+            ) as run_playbook_result:
+
+                # This checks if run_playbook returns a Process object (or more precisely: object returned from Popen)
+                self.assertEqual(run_playbook_result, popen_result)
+
+            mock_render.assert_called_once_with(
+                inventory_path='/tmp/string/file',
+                vars_path='/tmp/string/file',
+                requirements_path='/tmp/requirements.txt',
+                playbook_name='playbook_name',
+                remote_username='root',
+                venv_path='/tmp/tempdir/venv'
             )
-            self.assertEqual(
-                mock_popen.mock_calls,
-                [call(run_playbook_cmd, bufsize=1, stdout=-1, stderr=-1, cwd='/play/book', shell=True)]
+
+            mock_popen.assert_called_once_with(
+                "ANSIBLE CMD", bufsize=1, stdout=-1, stderr=-1, cwd='/play/book', shell=True, env=mock.ANY
             )
+
+    def test_render_command(self):
+        """
+        Run the render_sandbox_creation_command function
+        """
+
+        run_playbook_command = ansible.render_sandbox_creation_command(
+            requirements_path='/requirements/path.txt',
+            inventory_path="/tmp/inventory/path",
+            vars_path="/tmp/vars/path",
+            playbook_name='playbook_name',
+            remote_username="root",
+            venv_path='/tmp/venv'
+        )
+        expected = (
+            'virtualenv -p /usr/bin/python /tmp/venv && '
+            '/tmp/venv/bin/python -u /tmp/venv/bin/pip install -r /requirements/path.txt && '
+            '/tmp/venv/bin/python -u /tmp/venv/bin/ansible-playbook -i /tmp/inventory/path '
+            '-e @/tmp/vars/path -u root playbook_name'
+        )
+
+        self.assertEqual(expected, run_playbook_command)
+
+    def test_create_temp_dir_ok(self):
+        """
+        Check if create_temp_dir behaves correctly when no exception is
+        raised from inside it.
+
+        By behave correctly we mean: creates and returns a tempdir, which
+        is deleted after we exit the context manager.
+        """
+        with ansible.create_temp_dir() as temp_dir:
+            self.assertTrue(os.path.exists(temp_dir))
+            self.assertTrue(os.path.isdir(temp_dir))
+        self.assertFalse(os.path.exists(temp_dir))
+
+    def test_create_temp_dir_exception(self):
+        """
+        Check if create_temp_dir behaves correctly when an exception is
+        raised from inside it.
+
+        By behave correctly we mean: creates and returns a tempdir, which
+        is deleted after we exit the context manager.
+        """
+        saved_temp_dir = None
+        try:
+            with ansible.create_temp_dir() as temp_dir:
+                saved_temp_dir = temp_dir
+                self.assertTrue(os.path.exists(temp_dir))
+                self.assertTrue(os.path.isdir(temp_dir))
+                raise KeyboardInterrupt()
+        except KeyboardInterrupt:
```

Comment:
Would it be worth checking that the exception propagates outside the `with` block here? If someone makes a change in the future that catches the exception and doesn't re-raise it, the assertion here will be skipped and the test will pass even if the temp dir still exists.